File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 7/16/2026, 4:39:33 PM
class Solution {
public:
    bool isPalindrome(int x){
        if(x<0 || (x % 10 == 0 && x != 0)){
            return false;
        }
        int rev_half =0;

        while(x> rev_half){
            int digit = x % 10;
            rev_half = (rev_half * 10) + digit;
            x /= 10;
        }
        return x == rev_half || x == rev_half / 10;
    }
};

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 8/26/2026, 10:04:20 AM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        num = x
        result = 0
        while(num>0):
            id = (num%10)
            result = (result * 10) + id
            num = num // 10

        return result == x